### PR TITLE
rebase: update minikube to latest version

### PR DESCRIPTION
**Pull request by Madhu-1**
As minikube [1.17.1](https://github.com/kubernetes/minikube/releases) is released and updating the minikube to the latest available version.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: build.env [hunk 1]
```diff
@@ -36,7 +36,7 @@ SNAPSHOT_VERSION=v3.0.1
 HELM_VERSION=v3.1.2
 
 # minikube settings
-MINIKUBE_VERSION=v1.14.1
+MINIKUBE_VERSION=v1.16.0
```
Comment (obnoxxx) [inline build.env:39]: why not use `latest`?

Comment (Madhu-1) [reply build.env:39]: @obnoxxx currently, we are sticking to the available release once the version is tested in the CI.  if there is any regression in the latest available release we might end up in the CI issues which could block the merging of the PR.

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@Mergifyio rebase

The logs of the CI jobs have been removed, we will need new logs in order to fix issues.

**Comment by Madhu-1** [comment]
@Mergifyio rebase

The logs of the CI jobs have been removed, we will need new logs in order to fix issues.

**Comment by nixpanic** [comment]
https://github.com/kubernetes/minikube/tree/v1.17.1 has been released and includes a fix for #1840

**Review by nixpanic — DISMISSED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
The e2e tests seem to fail consistently with the following error:
```
Feb  1 09:40:18.655: INFO: Waiting up to 10m0s for all daemonsets in namespace 'cephcsi-e2e-3b6671624651' to start
Feb  1 09:40:18.658: INFO: 1 / 1 pods ready in namespace 'cephcsi-e2e-3b6671624651' in daemonset 'csi-cephfsplugin' (0 seconds elapsed)
[1mSTEP[0m: check static PVC
Feb  1 09:40:18.665: INFO: ExecWithOptions {Command:[/bin/sh -c ceph fsid] Namespace:rook-ceph PodName:rook-ceph-tools-5455675849-qxgb2 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Feb  1 09:40:18.665: INFO: >>> kubeConfig: /root/.kube/config
Feb  1 09:40:21.320: INFO: ExecWithOptions {Command:[/bin/sh -c ceph fs subvolumegroup create myfs testGroup] Namespace:rook-ceph PodName:rook-ceph-tools-5455675849-qxgb2 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Feb  1 09:40:21.320: INFO: >>> kubeConfig: /root/.kube/config
Feb  1 09:53:13.588: INFO: stdErr occurred: Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 165, in get_fs_handle
    conn.connect()
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 88, in connect
    self.fs.mount(filesystem_name=self.fs_name.encode('utf-8'))
  File "cephfs.pyx", line 739, in cephfs.LibCephFS.mount
cephfs.Error: error calling ceph_mount: Connection timed out [Errno 110]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1177, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/volumes/module.py", line 426, in handle_command
    return handler(inbuf, cmd)
  File "/usr/share/ceph/mgr/volumes/module.py", line 34, in wrap
    return f(self, inbuf, cmd)
  File "/usr/share/ceph/mgr/volumes/module.py", line 452, in _cmd_fs_subvolumegroup_create
    uid=cmd.get('uid', None), gid=cmd.get('gid', None))
  File "/usr/share/ceph/mgr/volumes/fs/volume.py", line 480, in create_subvolume_group
    with open_volume(self, volname) as fs_handle:
  File "/lib64/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 316, in open_volume
    fs_handle = vc.connection_pool.get_fs_handle(volname)
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 171, in get_fs_handle
    raise VolumeException(-e.args[0], e.args[1])
TypeError: bad operand type for unary -: 'str'


Feb  1 09:53:13.588: FAIL: failed to validate CephFS static pv with error command terminated with exit code 22
```

Possibly minikube has tightened its network policy and the node-plugin can not connect to the MDS anymore? During e2e testing, we use two different namespaces for the Ceph cluster and to-test ceph-csi services. Maybe that is problematic...

**Comment by Madhu-1** [comment]
@nixpanic something has changed in minikube 1.17.1 am not able to run ceph fs commands from the toolbox pod.

**Comment by nixpanic** [comment]
> @nixpanic something has changed in minikube 1.17.1 am not able to run ceph fs commands from the toolbox pod.

The toolbox pod can access the CephFS MDS, I think. Commands like `ceph fs status` work just fine.
However, when trying to create a subvolume group, the `ceph` command needs to talk to the CephMgr. Connecting to the CephMgr works as well, the logs of the CephMgr pod show that the command is received.
```
debug 2021-02-05T08:46:23.577+0000 7fc29d02b700  0 log_channel(audit) log [DBG] : from='client.11024 -' entity='client.admin' cmd=[{"prefix": "fs subvolumegroup create", "vol_name": "myfs", "target": ["mgr", ""], "group_name": "testGroup"}]: dispatch
```

However, it seems to run into a timeout:
```
debug 2021-02-05T08:50:14.554+0000 7fc29c82a700  0 [volumes ERROR volumes.module] Failed _cmd_fs_subvolumegroup_create(group_name:testGroup, prefix:fs subvolumegroup create, target:['mgr', ''], vol_name:myfs) < "":
Traceback (most recent call last):
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 165, in get_fs_handle
    conn.connect()
  File "/usr/share/ceph/mgr/volumes/fs/operations/volume.py", line 88, in connect
    self.fs.mount(filesystem_name=self.fs_name.encode('utf-8'))
  File "cephfs.pyx", line 739, in cephfs.LibCephFS.mount
cephfs.Error: error calling ceph_mount: Connection timed out [Errno 110]
```

Testing communications between toolbox, mds, mgr does not show any restrictions. Installed `nc` inside the running containers, and run is like this:

```
[root@rook-ceph-mgr-a-799d4886bd-4gmcs /]# nc -l -p 6902 --verbose
Ncat: Version 7.70 ( https://nmap.org/ncat )
Ncat: Listening on :::6902
Ncat: Listening on 0.0.0.0:6902
Ncat: Connection from 172.17.0.15.
Ncat: Connection from 172.17.0.15:49760.
ping
```

```
[root@rook-ceph-mds-myfs-a-f474f4b68-c8xxh /]# nc --verbose 172.17.0.10 6902 <<< "ping"
Ncat: Version 7.70 ( https://nmap.org/ncat )
Ncat: Connected to 172.17.0.10:6902.
Ncat: 5 bytes sent, 0 bytes received in 0.03 seconds.
```

I have not been able to identify the issue when running `ceph fs subvolumegroup create myfs testGroup`, but using the `--cni=bridge` parameter when starting minikube makes things work.

**Review by nixpanic — APPROVED** [review]

**Review by humblec — DISMISSED** [review]

**Comment by nixpanic** [comment]
**ci/centos/mini-e2e/k8s-1.20** failed with:
```
Feb  9 10:20:21.798: INFO: csi-cephfs-demo-pod app  to be deleted (600 seconds elapsed)
Feb  9 10:20:21.798: FAIL: failed to validate CephFS pvc and application  binding with error timed out waiting for the condition
```

Not sure what the cause is, [logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.20/runs/195/nodes/85/log/).

**Comment by nixpanic** [comment]
**ci/centos/mini-e2e-helm/k8s-1.20** timed out after running an hour [job 189](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.20/detail/mini-e2e-helm_k8s-1.20/189/pipeline). It was running fine and progressing... Maybe we need to increase the runtime of the e2e tests?

**Comment by Madhu-1** [comment]
> **ci/centos/mini-e2e/k8s-1.20** failed with:
> 
> ```
> Feb  9 10:20:21.798: INFO: csi-cephfs-demo-pod app  to be deleted (600 seconds elapsed)
> Feb  9 10:20:21.798: FAIL: failed to validate CephFS pvc and application  binding with error timed out waiting for the condition
> ```
> 
> Not sure what the cause is, [logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.20/runs/195/nodes/85/log/).

Analysis
```
I0209 10:10:23.315530       1 utils.go:132] ID: 42 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC call: /csi.v1.Node/NodeUnpublishVolume
I0209 10:10:23.315688       1 utils.go:133] ID: 42 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC request: {"target_path":"/var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount","volume_id":"0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3"}
I0209 10:14:06.241861       1 cephcmds.go:59] ID: 42 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 command succeeded: umount [/var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount]
I0209 10:14:06.242222       1 nodeserver.go:277] ID: 42 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 cephfs: successfully unbinded volume 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 from /var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount
I0209 10:14:06.242277       1 utils.go:138] ID: 42 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC response: {}
```

It took around 4 minutes to umount the targetPath after that the NodePublish is failing with below error

```log
I0209 10:14:31.025118       1 utils.go:132] ID: 56 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC call: /csi.v1.Node/NodeUnpublishVolume
I0209 10:14:31.025255       1 utils.go:133] ID: 56 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC request: {"target_path":"/var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount","volume_id":"0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3"}
I0209 10:14:31.031176       1 cephcmds.go:53] ID: 56 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 an error (exit status 32) occurred while running umount args: [/var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount]
E0209 10:14:31.031215       1 utils.go:136] ID: 56 Req-ID: 0001-0024-c3feb7b7-cf6b-4154-b66f-fb09df8d20b7-0000000000000001-fc81f537-6abe-11eb-9ad3-e6ef196d01c3 GRPC error: rpc error: code = Internal desc = an error (exit status 32) occurred while running umount args: [/var/lib/kubelet/pods/9d150e08-6b25-478d-96a6-03bfd96f907a/volumes/kubernetes.io~csi/pvc-2a3650cd-8454-40e8-ae4b-31a7f6034dec/mount]
```
as the umount is already done cephcsi should return success not the Internal server error.

**Comment by nixpanic** [comment]
ci/centos/mini-e2e/k8s-1.20 failed with a timeout in the test suite again:
```
Feb  9 11:52:47.562: INFO: rbd-32742 app  is in Pending phase expected to be in Running  state (8 seconds elapsed)

panic: test timed out after 1h0m0s
```

Will increase the timeout to 90 minutes.

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
ci/centos/mini-e2e-helm/k8s-1.20 failed with some unexpected error:
```
Feb  9 13:15:34.629: INFO: Deleting PersistentVolumeClaim csi-cephfs-pvc on namespace cephfs-3274
Feb  9 13:15:34.645: INFO: waiting for PVC csi-cephfs-pvc in state &PersistentVolumeClaimStatus{Phase:Bound,AccessModes:[ReadWriteMany],Capacity:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},Conditions:[]PersistentVolumeClaimCondition{},} to be deleted (0 seconds elapsed)
Feb  9 13:15:36.684: INFO: waiting for PVC csi-cephfs-pvc in state &PersistentVolumeClaimStatus{Phase:Bound,AccessModes:[ReadWriteMany],Capacity:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},Conditions:[]PersistentVolumeClaimCondition{},} to be deleted (2 seconds elapsed)
[AfterEach] cephfs
  /go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/framework.go:175
Feb  9 13:15:36.690: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
```

[logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e-helm_k8s-1.20/detail/mini-e2e-helm_k8s-1.20/194/pipeline)

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e/k8s-1.20

Failed due to #1795

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.20

Resizing a CephFS PVC failed:
```
Feb  9 15:42:41.257: FAIL: failed to resize PVC with error timed out waiting for the condition
...
Feb  9 15:42:41.271: INFO: At 2021-02-09 15:21:29 +0000 GMT - event for cephfs-32740: {kubelet minikube} FailedMount: MountVolume.MountDevice failed for volume "pvc-75f9da6d-8b2e-431c-a5a0-3a191339f8fe" : rpc error: code = DeadlineExceeded desc = context deadline exceeded
Feb  9 15:42:41.271: INFO: At 2021-02-09 15:21:30 +0000 GMT - event for cephfs-32740: {kubelet minikube} FailedMount: MountVolume.MountDevice failed for volume "pvc-75f9da6d-8b2e-431c-a5a0-3a191339f8fe" : rpc error: code = Aborted desc = an operation with the given Volume ID 0001-0024-f047ded8-7850-46dd-abc2-87233fa3069d-0000000000000001-30e2817d-6aea-11eb-a6c4-863f529beb9b already exists
...
```
Maybe the provisioner got hung or something, but I did not immediately see it in [the logs](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e-helm_k8s-1.20/runs/199/nodes/96/log).

**Comment by Madhu-1** [comment]
@nixpanic looks like merging this PR can affect the cephfs e2e testing. When looking at multiple logs, if feel now cephfs is taking a lot of time for each operation. Merging the PR will make the CI flaky.

**Comment by Madhu-1** [comment]
@mergifyio  rebase

**Comment by Madhu-1** [comment]
@Mergifyio rebase

File: scripts/minikube.sh [hunk 2]
```diff
@@ -162,6 +162,7 @@ CONTAINER_CMD=${CONTAINER_CMD:-"docker"}
 MEMORY=${MEMORY:-"4096"}
 CPUS=${CPUS:-"$(nproc)"}
 VM_DRIVER=${VM_DRIVER:-"virtualbox"}
+CNI=${CNI:-"bridge"}
```
Comment (humblec) [inline scripts/minikube.sh:165]: @nixpanic whats the default CNI ? if its not `bridge` may be the change for CNI to `bridge` causing the E2E to take lots of time to complete , iow, we are hitting some performance issues being the CNI bridge

Comment (nixpanic) [reply scripts/minikube.sh:165]: The default CNI is "auto"... There seems to be some logic in minikube somewhere that decides what CNI to use (maybe dependent on the Kubernetes version and hypervisor?).

It is possible that the minikube VM needs more resources with this rebase. Those are settings in the `ci/centos` branch, so we could increase those, depending on the current values and bare metal systems in the CI.

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e-helm/k8s-1.20

Failed to delete a volume, the PVC that needs to be deleted seems to be gone. Looks like a bug in the test case:
```
Feb 10 07:28:36.559: INFO: waiting for PVC rbd-13182 in state &PersistentVolumeClaimStatus{Phase:,AccessModes:[],Capacity:ResourceList{},Conditions:[]PersistentVolumeClaimCondition{},} to be deleted (600 seconds elapsed)
Feb 10 07:28:36.564: INFO: failed to delete PVC and application (rbd-13182): timed out waiting for the condition
Feb 10 07:28:36.564: FAIL: deleting PVCs and applications failed, 1 errors were logged
```

**Review by Madhu-1 — COMMENTED** [review]
LGTM

**Pull request merged**